onnxruntime/core/session/inference_session.h

```diff
@@ -135,7 +136,12 @@ class InferenceSession {
     If nullptr, the default LoggingManager MUST have been created previously as it will be used
     for logging. This will use the default logger id in messages.
     See core/common/logging/logging.h for details, and how LoggingManager::DefaultLogger works.
+	@param thread_pool If it's NULL, the session will run as single threaded.
     */
+  InferenceSession(const SessionOptions& session_options,
+                   logging::LoggingManager* logging_manager,
+                   concurrency::ThreadPool* thread_pool);
+
   explicit InferenceSession(const SessionOptions& session_options,
```

Comment:
This may confusing users. 

When writing codes as InferenceSession(session_option), it's using thread pool (not single thread), but when writing codese as InferneceSession(session_option, null, null), it's single thread.

Reply:
When writing codes as InferenceSession(session_option) and session_options.thread_count=1,it's using thread pool (not single thread). Otherwise it's single threaded.

Reply:
No matter which function you use,
1. if you pass in a thread pool, the session will use it
2. otherwise, it will create a new one based on session options.

Reply:
I'll revert this change. We can discuss this later.